wgpu/src/backend/webgpu.rs

```diff
@@ -3808,3 +3816,32 @@ impl Drop for BufferMappedRange {
         }
     }
 }
+
+/// Adds the constants map to the given pipeline descriptor if the map is nonempty.
+/// Logs an error if the map cannot be set.
+///
+/// This function is necessary because the constants array is not currently
+/// exposed by `wasm-bindgen`. See the following issues for details:
+/// - [gfx-rs/wgpu#5688](https://github.com/gfx-rs/wgpu/pull/5688)
+/// - [rustwasm/wasm-bindgen#3587](https://github.com/rustwasm/wasm-bindgen/issues/3587)
+fn insert_constants_map(target: &JsValue, map: &HashMap<String, f64>) {
+    if !map.is_empty() {
+        let result = js_sys::Reflect::set(target, &"constants".into(), &hashmap_to_jsvalue(map));
+
+        if let Err(error) = result {
+            log::error!("Failed to set constants map for pipeline: {error:?}");
+        }
```

Comment:
```suggestion
        js_sys::Reflect::set(target, &"constants".into(), &hashmap_to_jsvalue(map))
            .expect("Setting the values in a Javascript object should never fail");
```

I don't think we need to deal with the error case.

Reply:
not a very strong stance here but I'd rather we don't do `expect` and keep the `log` (I'm guilty of suggesting this earlier on the PR). Is it likely this ever breaks? No, very very unlikely! But an `expect` already implies checking on the error (so does unwrap) and I'd rather not have us unrecoverable crash over this ever.

(and yes I hedge similar concerns about _a lot_ of code in wgpu and wgpu-core ;). If we disagree a lot here maybe we should talk this out at some point to have more unified stance on this stuff going forward. But again this specific instance is like one of the least concerning ones, so just a leaning here 🤷 )

Reply:
> But an expect already implies checking on the error (so does unwrap) and I'd rather not have us unrecoverable crash over this ever.

I would much rather an expect than a log that everyone will ignore - more specifically, this code should not _ever_ fail unless something horribly wrong with the js reflect api exists. I agree we shouldn't unwrap if there's a chance for the user to want to fail-and-try-again.